jacs-model-access/src/main/java/org/janelia/model/access/domain/dao/mongo/TmWorkspaceMongoDao.java

```diff
@@ -88,6 +97,64 @@ public List<TmWorkspace> getAllTmWorkspaces(String subjectKey) {
                 TmWorkspace.class);
     }
 
+    @Override
+    public List<TmWorkspaceInfo> getLargestWorkspaces(String subjectKey, Long limit) {
+        // Step 1: Get all accessible workspaces using existing method
+        List<TmWorkspace> workspaces = getAllTmWorkspaces(subjectKey);
+
+        List<TmWorkspaceInfo> workspaceInfoList = new ArrayList<>();
```

Comment:
Why is this new model necessary? Why not just return the TmWorkspace objects themselves? That would make this method more generally usable, and it wouldn't require expanding the model.